src/client/java/teammates/client/scripts/DataMigrationForFeedbackResponseCommentSearchDocumentDateFormat.java

```diff
@@ -0,0 +1,166 @@
+
+            String sampleDateString = extractSampleDateString(document);
+            if (isInDateFormat(sampleDateString, oldDateFormat)) {
+                commentsToFix.add(comment);
+            } else if (isInDateFormat(sampleDateString, newDateFormat)) {
+                numberOfUnaffectedDocuments++;
+            } else {
+                println("Unrecognised date format (" + sampleDateString + ") for:\n" + comment);
+            }
+        }
+
+        println("\n################# Analysis Results #################");
+        println("Total number of comments: " + loopHelper.getCount());
+        println("Total number of documents: " + numberOfDocuments + "\n");
+
+        println("Number of documents already in new date format: " + numberOfUnaffectedDocuments);
+        println("Number of documents in unrecognizable date format: "
+                + (numberOfDocuments - commentsToFix.size() - numberOfUnaffectedDocuments) + "\n");
+
+        println("Number of documents in old date format: " + commentsToFix.size());
+        println("####################################################\n");
+
+        return commentsToFix;
+    }
+
+    private Document getDocument(FeedbackResponseCommentAttributes comment) {
+        return index.get(comment.getId().toString());
+    }
+
+    private String extractSampleDateString(Document document) {
+        String frcaJson = document.getOnlyField(Const.SearchDocumentField.FEEDBACK_RESPONSE_COMMENT_ATTRIBUTE).getText();
+        return jsonParser.parse(frcaJson).getAsJsonObject().getAsJsonPrimitive("createdAt").getAsString();
+    }
+
+    private synchronized boolean isInDateFormat(String dateString, DateFormat dateFormat) {
+        try {
+            dateFormat.parse(dateString);
+            return true;
+        } catch (ParseException e) {
+            return false;
+        }
+    }
+
+    private void fixDocuments(List<FeedbackResponseCommentAttributes> commentsToFix) {
+        println("Running data migration for " + commentsToFix.size() + " documents in old date format...");
+        println("Preview: " + isPreview);
+
+        List<Document> documentsToUpdate = new ArrayList<>();
+
+        LoopHelper loopHelper = new LoopHelper(batchSize, "documents processed.");
+
+        for (FeedbackResponseCommentAttributes commentToFix : commentsToFix) {
+            loopHelper.recordLoop();
+            queueDocumentUpdate(commentToFix, documentsToUpdate);
+
+            if (documentsToUpdate.size() == batchSize) {
+                updateAndClearDocuments(documentsToUpdate);
+            }
+        }
+        updateAndClearDocuments(documentsToUpdate);
+
+        println("\nComplete! If there are any errors shown above, please rerun this script.");
+    }
+
+    private void queueDocumentUpdate(FeedbackResponseCommentAttributes comment, List<Document> documentsToUpdate) {
+        documentsToUpdate.add(new FeedbackResponseCommentSearchDocument(comment).build());
+    }
```

Comment:
I feel that it would be better to inline this method, since it is relatively simple. What do you think?